libs/vkd3d/command.c

```diff
@@ -2693,38 +2780,27 @@ static void d3d12_command_list_update_packed_descriptors(struct d3d12_command_li
         base_descriptor = d3d12_desc_from_gpu_handle(bindings->descriptor_tables[root_parameter_index]);
 
         table = root_signature_get_descriptor_table(root_signature, root_parameter_index);
-        vk_descriptor = &updates->descriptors[table->first_packed_descriptor];
-
         table_offsets[table->table_index] = d3d12_desc_heap_offset(base_descriptor);
 
-        for (i = 0; i < table->binding_count; i++)
+        if (deferred_update)
         {
-            const struct vkd3d_shader_resource_binding *binding = &table->first_binding[i];
-
-            if (binding->flags & VKD3D_SHADER_BINDING_FLAG_BINDLESS)
-                continue;
-
-            for (j = 0; j < binding->register_count; j++)
-            {
-                const struct d3d12_desc *desc = &base_descriptor[binding->descriptor_offset + j];
-
-                /* Skip invalid descriptors */
-                if (vkd3d_descriptor_info_from_d3d12_desc(list->device, desc, binding, vk_descriptor))
-                {
-                    vk_write_descriptor_set_for_descriptor_info(descriptor_set, binding->binding.binding,
-                            desc->vk_descriptor_type, vk_descriptor, &updates->descriptor_writes[write_count++]);
-                }
-
-                vk_descriptor++;
-            }
+            /* If we have EXT_descriptor_indexing we implement RS 1.0 correctly by deferring the descriptor
+             * set update until submit time. */
+            d3d12_command_list_defer_update_descriptor_table(list,
+                    descriptor_set, updates,
+                    root_signature, base_descriptor,
+                    root_parameter_index);
+        }
+        else
+        {
+            /* Fallback, we update the descriptor set here.
+             * Will work in most cases, but it's not a correct implementation of RS 1.0.
+             * Will work if application uses RS 1.1 and declares STATIC descriptor flag. */
```

Comment:
I guess we should respect that flag, might be worth adding a TODO for that somewhere. Currently we only have access to RS1.0 data structures though, so that will need some more work.